### ISSUE TITLE
MAINTAINED FORK: K-and-R/easy_captcha (gem `kandr-easy_captcha`)

### ISSUE THREAD

**Issue by karlwilbur** [phatworx/easy_captcha#36]
I have forked this and am working to improve its code quality and test coverage, as well as address some of the issues. 

https://github.com/K-and-R/easy_captcha

https://rubygems.org/gems/kandr-easy_captcha

As my fork is a fork, I cannot create issues there. I wish to keep it as a connected fork rather than delete the fork and recreate it as a separately cloned repo. 

Should anyone have any changes that are needed, please feel free to tag me or message directly.

**Comment by kopylovvlad** [comment]
Great idea! Previously I created a fork with Rails5 support [link](https://github.com/phatworx/easy_captcha/pull/33). But unfortunately, the original maintainer of this gem is not involved anymore.

**Comment by karlwilbur** [comment]
Yes, I went through and merged many of the outstanding pull requests here. I merged your PR here: https://github.com/K-and-R/easy_captcha/commit/664bac20b1c0dcc0a3a41a63b15fd5229823bbe5

**Comment by kopylovvlad** [comment]
Awesome! Do you have troubles with rmagick version 4.2.2? Today I merged [little improvement](https://github.com/kopylovvlad/easy_captcha/pull/2) from @BBenni

**Comment by karlwilbur** [comment]
I don't recall specifically having trouble with RMagick. I did merge in quite a few different things and made several update/changes of my own. Tests are passing and I've been working on adding more.

**Comment by BBenni** [comment]
Yes the problem does not arise in the K-and-R fork as it uses a different implementation at the method in question than your fork @kopylovvlad .

In general: Thanks a lot for your work @kopylovvlad and @karlwilbur!

**Comment by BBenni** [comment]
K-and-R fork works very fine for me as is, except if I use caching.

If I use caching I get the following error:
```
 Exception: ArgumentError: wrong number of arguments (given 1, expected 0)
/usr/local/bundle/gems/kandr-easy_captcha-0.11.0/lib/easy_captcha/controller_helpers.rb:35:in `captcha_cache_path'
```

After a quick look it seems that there is indeed a problem in `ControllerHelpers`:

In the method `cache_image` you are calling `captcha_cache_path` with argument `code` (https://github.com/K-and-R/easy_captcha/blob/master/lib/easy_captcha/controller_helpers.rb#L130).

`captcha_cache_path` does however not take any arguments (https://github.com/K-and-R/easy_captcha/blob/master/lib/easy_captcha/controller_helpers.rb#L35).

**Comment by karlwilbur** [comment]
@BBenni Thanks. I'll look into that.
